### PR TITLE
Switch resource processor to use MSI

### PR DESCRIPTION
**Pull request by marrobi**
# PR for issue 457

## What is being addressed

Configure resource processor to use MSI

## How is this addressed

- Add Owner rights to VMMS MSI, See for future: #495 
- Update runner.py to set client_secret to blank value if not set
- Remove porter_driver parameters

### CODE REVIEW

File: processor_function/vm_porter/runner.py [hunk 1]
```diff
@@ -188,11 +189,19 @@ def read_env_vars():
...
         # Needed for running porter
+        "arm_use_msi": os.environ["ARM_USE_MSI"],
         "arm_subscription_id": os.environ['ARM_SUBSCRIPTION_ID'],
         "arm_client_id": os.environ["ARM_CLIENT_ID"],
-        "arm_client_secret": os.environ["ARM_CLIENT_SECRET"],
-        "arm_tenant_id": os.environ["ARM_TENANT_ID"],
+        "arm_tenant_id": os.environ["ARM_TENANT_ID"]
     }
+
+    if env_vars["arm_use_msi"] == "false":
```
Comment (sachinkundu) [inline processor_function/vm_porter/runner.py:198]: env_vars["arm_client_secret"]  = os.environ["ARM_CLIENT_SECRET"] if env_vars["arm_use_msi"] == "false" else "not_required"

File: processor_function/.env.sample [hunk 2]
```diff
@@ -28,4 +28,9 @@ SEC_ARM_CLIENT_ID=__CHANGE_ME__
 SEC_ARM_CLIENT_SECRET=__CHANGE_ME__
 param_tfstate_resource_group_name=__CHANGE_ME__
 param_tfstate_container_name=__CHANGE_ME__
-param_tfstate_storage_account_name=__CHANGE_ME__
\ No newline at end of file
+param_tfstate_storage_account_name=__CHANGE_ME__
+
+# The below are needed for local development of the VMSS resource processor
+ARM_USE_MSI=false
+RESOURCE_PROCESSOR_CLIENT_ID=__CHANGE_ME__
```
Comment (sachinkundu) [inline processor_function/.env.sample:35]: Is this really still needed for VMSS processor when working locally? I think we can use local login credentials to do even porter deployments as long as the user has the right set of permissions.

File: processor_function/vm_porter/runner.py (hunk 1 above)
Comment (marrobi) [reply processor_function/vm_porter/runner.py:198]: trying to get head round that - so thats a single line? any docs/python 101 I can read?!

File: processor_function/.env.sample (hunk 2 above)
Comment (marrobi) [reply processor_function/.env.sample:35]: Its the TF inside the bundles that needs them. As the bundles run inside a container, can't see how would access the local creds?

**Review by sachinkundu — APPROVED** [review]

**Pull request merged**